var/spack/repos/builtin/packages/py-mkdocs/package.py

```diff
@@ -14,17 +14,29 @@ class PyMkdocs(PythonPackage):
     homepage = "https://www.mkdocs.org/"
     pypi = "mkdocs/mkdocs-1.3.1.tar.gz"
 
+    version("1.4.2", sha256="8947af423a6d0facf41ea1195b8e1e8c85ad94ac95ae307fe11232e0424b11c5")
     version("1.3.1", sha256="a41a2ff25ce3bbacc953f9844ba07d106233cd76c88bac1f59cb1564ac0d87ed")
 
     depends_on("python@3.6:", type=("build", "run"))
-    depends_on("py-setuptools", type="build")
+    depends_on("python@3.7:", when="@1.4.2:", type=("build", "run"))
+    depends_on("py-setuptools", when="@:1.3", type="build")
+    depends_on("py-hatchling", when="@1.4.2:", type="build")
     depends_on("py-click@3.3:", type=("build", "run"))
+    depends_on("py-click@7.0:", when="@1.4.2:", type=("build", "run"))
     depends_on("py-jinja2@2.10.2:", type=("build", "run"))
+    depends_on("py-jinja2@2.11.1:", when="@1.4.2:", type=("build", "run"))
     depends_on("py-markdown@3.2.1:3.3", type=("build", "run"))
     depends_on("py-pyyaml@3.10:", type=("build", "run"))
+    depends_on("py-pyyaml@5.1:", when="@1.4.2:", type=("build", "run"))
     depends_on("py-watchdog@2.0:", type=("build", "run"))
     depends_on("py-ghp-import@1.0:", type=("build", "run"))
     depends_on("py-pyyaml-env-tag@0.1:", type=("build", "run"))
-    depends_on("py-importlib-metadata@4.3:", type=("build", "run"))
+    depends_on("py-importlib-metadata@4.3:", when="@1.3.1", type=("build", "run"))
+    depends_on("py-importlib-metadata@4.3:", when="@1.4.2: ^python@:3.9", type=("build", "run"))
+    depends_on("py-typing-extensions@3.10:", when="@1.4.2: ^python@:3.7", type=("build", "run"))
     depends_on("py-packaging@20.5:", type=("build", "run"))
     depends_on("py-mergedeep@1.3.4:", type=("build", "run"))
+    depends_on("py-colorama@0.4:", when="platform=windows", type=("build", "run"))
+    depends_on("py-babel@2.9.0:", when="@1.4.2:", type=("build", "run"))
+
+    # py-babel is actually optional for mkdocs, but when installing with spack without the optional dependency, the build fails...
```

Comment:
What's the error message?

Reply:
This isthe one where we talked about it earlier in the code review process: 

https://github.com/spack/spack/pull/33672#discussion_r1021696047

Reply:
> Can you add a comment to this effect? Something like "This dep is not optional, build fails without it". This will help me a year from now when someone tries to update this package and I assume the dep is optional.

This comment didn't even help me 9 months later. How about:
```suggestion
    # Babel is optional according to pyproject.toml, but is imported in hatch_build.py
    # Build fails without it, so make it required
```

Reply:
I would also put the babel dep below this comment so it's clear that it's separate